src/main/java/bio/terra/cli/utils/MountUtils.java

```diff
@@ -0,0 +1,241 @@
+        resourceMountPaths.put(resource.getId(), mountPath);
+      } else {
+        resourceMountPaths.put(resource.getId(), WORKSPACE_DIR.resolve(resource.getName()));
+      }
+    }
+    return resourceMountPaths;
+  }
+
+  /**
+   * Helper method to get mountable resources in the workspace, any file containing resource or
+   * reference to a file containing resource.
+   */
+  public static List<Resource> getMountableResources() {
+    return Context.requireWorkspace().listResources().stream()
+        .filter(
+            r -> {
+              if (r.getResourceType() == Resource.Type.GCS_BUCKET) {
+                return true;
+              } else if (r.getResourceType() == Resource.Type.GCS_OBJECT) {
+                try {
+                  return ((GcsObject) r).isPrefix();
+                } catch (SystemException e) {
+                  // Pass through GCS objects that are inaccessible to display error on mounted
+                  // folder later
+                  return true;
+                }
+              }
+              return false;
+            })
+        .toList();
+  }
+
+  /**
+   * Helper method to get workspace folder paths. Used to build the mount paths for mountable
+   * resources.
+   *
+   * @return A map of folder IDs to folder paths.
+   */
+  private static Map<UUID, Path> getFolderPaths() {
+    List<Folder> folders = Context.requireWorkspace().listFolders();
+    Map<UUID, Path> folderPaths = new HashMap<>();
+
+    for (Folder folder : folders) {
+      Path path = Paths.get(folder.getDisplayName());
+
+      // Recursively prepend parent folder names to path
+      UUID parentId = folder.getParentFolderId();
+      while (parentId != null) {
+        // Get parent folder
+        Folder parent = null;
+        for (Folder f : folders) {
+          if (f.getId().equals(parentId)) {
+            parent = f;
+            break;
+          }
+        }
+        // Prepend parent folder name to path
+        if (parent != null) {
+          path = Paths.get(parent.getDisplayName()).resolve(path);
+          parentId = parent.getParentFolderId();
+        }
+      }
+      folderPaths.put(folder.getId(), path);
+    }
+    return folderPaths;
+  }
+
+  /**
+   * Creates local directories for all the given resource paths.
+   *
+   * @param paths A list of resource paths to create directories for.
+   * @throws SystemException If the creation of the directories failed.
+   */
+  public static void createResourceDirectories(List<Path> paths) throws SystemException {
+    for (Path path : paths) {
+      if (!path.toFile().exists()) {
+        if (!path.toFile().mkdirs()) {
+          throw new SystemException("Failed to create directory: " + path);
+        }
+      }
+    }
+  }
+
+  /**
+   * Recursively deletes all empty subdirectories in the WORKSPACE_DIR, excluding the WORKSPACE_DIR
+   * itself.
+   *
+   * <p>Throws UserActionableException if a non-empty directory is encountered. Throws
```

Comment:
Thanks for pointing this out. So this error would most likely be thrown when the user already has non empty directories in `~/workspace` and the user tries to run `terra resource mount`. This is unlikely to happen on CE's when this command will run in post startup. The other rare case is if users manually mess with the mounts after already mounting, e.g. unmounting a resource and then adding files to the resource's directory, then this will get thrown on remount.

In either case, we expect that there are no local files`~/workspace` and I think we should bring the user's attention to them if there are.